subprojects/base-services/src/main/java/org/gradle/internal/operations/notify/BuildOperationFinishedNotification.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.operations.notify;
+
+public interface BuildOperationFinishedNotification {
```

Comment:
we should use the timestamps from the source events as well, rather than recreating them, so that we later have the option of delivering the notifications asynchronously (but still in order).